internal/corazawaf/rule.go

```diff
@@ -204,15 +229,37 @@ func (r *Rule) doEvaluate(phase types.RulePhase, tx *Transaction, cache map[tran
 	} else {
 		ecol := tx.ruleRemoveTargetByID[r.ID_]
 		for _, v := range r.variables {
-			// TODO(anuraaga): Support skipping variables based on phase for rule chains too.
-			if multiphaseEvaluation && !r.HasChain && r.ParentID_ == 0 {
+			if multiphaseEvaluation && r.ParentID_ == 0 && (!r.HasChain || phase >= r.chainMinPhase) {
 				min := minPhase(v.Variable)
 				// When multiphase evaluation is enabled, any variable is evaluated at its
-				// earliest possible phase, so we skip it if the rule refers to other phases
-				// to avoid double-evaluation.
-				if min != types.PhaseUnknown && min != phase {
-					continue
+				// earliest possible phase, so we make sure to skip in other phases.
+				if min != types.PhaseUnknown {
+					if !r.HasChain {
```

Comment:
nit: I usually find disturbing to do a negation and then an `else`. Shall we swap the cases?